src/device/socket/vsock.rs

```diff
@@ -101,24 +139,185 @@ impl<'a, H: Hal, T: Transport> VirtIOSocket<'a, H, T> {
             op: VirtioVsockOp::Request.into(),
             ..Default::default()
         };
-        self.tx
-            .add_notify_wait_pop(&[header.as_bytes(), &[]], &mut [], &mut self.transport)?;
+        self.send_packet_to_tx_queue(&header, &[])?;
```

Comment:
Please add a comment here explaining why you're sending an empty packet, as it seems a bit surprising.